scripts/asm-installer/asm_vm

```diff
@@ -1110,6 +1110,22 @@ this project, run:
   --project="${PROJECT_ID}"
 EOF
     fi
+
+    local INSTANCE_TEMPLATE_REQ=$(retry 2 run gcloud compute instance-templates \
+      describe "${SOURCE_INSTANCE_TEMPLATE}" \
+      --format=json \
+      --project="${PROJECT_ID}")
+    local source_image_path="$(echo "${INSTANCE_TEMPLATE_REQ}" \
+    | jq '.properties.disks[].initializeParams.sourceImage')"
+
+    # check whether it is a  public gcp image
+    if [[ "$source_image_path" == *projects* && "$source_image_path" == *global/images/* ]]; then
```

Comment:
We need to check the full path, e.g. projects/[debian|centos]-cloud/global/images (syntax may not be correct)